### DISCUSSION TITLE
Newbie question: CPU backend?

### DISCUSSION THREAD

**Discussion by saona-raimundo** [parasyte/pixels#254] [Q&A] [answered]
I know the aim of the crate is hardware acceleration, and maybe you resolve my question by redirecting me to another crate, but... is it possible to show pixels in the screen using only the CPU? 

Of course the answer has to be yes, as there were no gpus before... but how to do it in Rust is my question and if this crate will ever consider having a cpu backend.

Thank you beforehand!

**Comment by parasyte** [comment, marked as answer]
Your system already provides a CPU adapter, as mentioned in https://github.com/parasyte/pixels/issues/92#issuecomment-1008293371

See documentation for [`RequestAdapterOptions::force_fallback_adapter`](https://docs.rs/wgpu/0.12.0/wgpu/struct.RequestAdapterOptionsBase.html#structfield.force_fallback_adapter)

> Indicates that only a fallback adapter can be returned. This is generally a “software” implementation on the system.

And the output from [the `wgpu-info` cli tool](https://github.com/gfx-rs/wgpu/tree/v0.12/wgpu-info). (Look for "Type: Cpu")
